web-common/src/components/menu/LeaderboardMeasureNamesDropdown.svelte

```diff
@@ -0,0 +1,234 @@
+<script lang="ts">
+  import * as DropdownMenu from "@rilldata/web-common/components/dropdown-menu/";
+  import Tooltip from "@rilldata/web-common/components/tooltip/Tooltip.svelte";
+  import { fly } from "svelte/transition";
+  import CaretDownIcon from "@rilldata/web-common/components/icons/CaretDownIcon.svelte";
+  import TooltipContent from "@rilldata/web-common/components/tooltip/TooltipContent.svelte";
+  import Search from "@rilldata/web-common/components/search/Search.svelte";
+  import Button from "@rilldata/web-common/components/button/Button.svelte";
+  import type { MetricsViewSpecMeasure } from "@rilldata/web-common/runtime-client";
+  import Switch from "@rilldata/web-common/components/forms/Switch.svelte";
+  import InputLabel from "@rilldata/web-common/components/forms/InputLabel.svelte";
+  import { writable } from "svelte/store";
+
+  export let disabled = false;
+  export let searchText = "";
+  export let visibleMeasures: MetricsViewSpecMeasure[];
+  export let leaderboardSortByMeasureName: string;
+  export let selectedMeasureNames: string[];
+  export let setLeaderboardMeasureNames: (names: string[]) => void;
+  export let setLeaderboardSortByMeasureName: (name: string) => void;
+
+  let active = false;
+  let multiSelect = selectedMeasureNames?.length > 1;
+
+  const lastSelectedMeasures = writable<string[]>([]);
+
+  $: filteredMeasures = visibleMeasures
+    .filter((item) =>
+      ((item.displayName || item.name) ?? "")
+        .toLowerCase()
+        .includes(searchText.toLowerCase()),
+    )
+    .sort((a, b) => {
+      const aIndex = visibleMeasures.findIndex((m) => m.name === a.name);
+      const bIndex = visibleMeasures.findIndex((m) => m.name === b.name);
+      return aIndex - bIndex;
+    });
+
+  $: showingMeasuresText =
```

Comment:
I'm relying on the sort by measure if no leaderboard measure names are selected. It feels like the correct path is to always set the leaderboard measure names with the first visible measure, just like the default behavior of sort by measure

Reply:
https://github.com/rilldata/rill/pull/7098/commits/05507e723538afe07315230c85a7a724fc53b0f4